web/src/utils/generateInputs.ts

```diff
@@ -0,0 +1,164 @@
+import { secp256k1 } from '@noble/curves/secp256k1'
+import * as mod from '@noble/curves/abstract/modular'
+import * as utils from '@noble/curves/abstract/utils'
+import { hexToNumber, hexToBigInt } from 'viem'
+import { ProjPointType, SignatureType } from '@noble/curves/abstract/weierstrass'
+import { createTree, DefaultsForUserOp, executeTransactionData, calculatePrecomputes, addressToProjectivePoint, splitToRegisters, hasher, getUserOpHash, UserOperation } from 'common';
+import { IncrementalMerkleTree } from '@zk-kit/incremental-merkle-tree'
+
+let tree: IncrementalMerkleTree
+createTree(4, 0n, 2).then((t) => {
+  tree = t
+  console.log(tree)
+
+  const priv = new Uint8Array(32).fill(1)
+  const pub = secp256k1.getPublicKey(priv)
+  const publicKeyPoint = secp256k1.ProjectivePoint.fromHex(pub)
+  const secret = 1n
+
+  const Qa = [
+    ...splitToRegisters(publicKeyPoint.toAffine().x),
+    ...splitToRegisters(publicKeyPoint.toAffine().y),
+  ]
+
+  hasher([...Qa, secret]).then((nullifier) => {
+    tree.insert(BigInt(nullifier))
+  })
+})
+
+type Signer = {
+  priv: Uint8Array
+  pub: Uint8Array
+  publicKeyPoint: ProjPointType<bigint>
+  secret: bigint
+  nullifier: bigint
+}
+
+interface SignerWithTree extends Signer {
+  siblings: number[]
+  pathIndices: any[]
+}
+
+const isDebugMode = false;
+
+function assert(condition: boolean, message?: string) {
+  if (isDebugMode && !condition) {
+    throw new Error(message || 'Assertion failed');
+  }
+}
+
+export const generateInputs = async (
```

Comment:
At some point we should extract this out as a common utility for testing and the client, but this is good for now!

Reply:
True! Completely with you.